bridge/resources/thingDescription/resource.go

```diff
@@ -0,0 +1,114 @@
+/****************************************************************************
+ *
+ * Copyright (c) 2024 plgd.dev s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"),
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+package thingDescription
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/plgd-dev/device/v2/bridge/net"
+	"github.com/plgd-dev/device/v2/bridge/resources"
+	"github.com/plgd-dev/device/v2/schema"
+	"github.com/plgd-dev/device/v2/schema/interfaces"
+	"github.com/plgd-dev/go-coap/v3/message"
+	"github.com/plgd-dev/go-coap/v3/message/codes"
+	"github.com/plgd-dev/go-coap/v3/message/pool"
+	"github.com/web-of-things-open-source/thingdescription-go/thingDescription"
+)
+
+const (
+	ResourceURI  = "/.well-known/wot"
+	ResourceType = "wot.thing"
+)
+
+var MessageTypeTDJson = message.MediaType(432) ///< application/td+json
+
+type (
+	GetThingDescription  func(ctx context.Context, endpoints schema.Endpoints) *thingDescription.ThingDescription
+	RegisterSubscription func(subscription func(td *thingDescription.ThingDescription, closed bool)) func()
+)
+
+type Resource struct {
+	*resources.Resource
+	onGetThingDescription  GetThingDescription
+	onRegisterSubscription RegisterSubscription
+}
+
+func (r *Resource) createMessage(request *net.Request, thingDescription *thingDescription.ThingDescription) (*pool.Message, error) {
+	dataJson, err := json.Marshal(thingDescription)
+	if err != nil {
+		return resources.CreateErrorResponse(request.Context(), codes.InternalServerError, err)
+	}
+	mediaType, err := request.Accept()
+	if err != nil {
+		mediaType = message.AppJSON
+	}
+	switch mediaType {
+	case message.AppJSON, MessageTypeTDJson:
+		res := pool.NewMessage(request.Context())
+		res.SetCode(codes.Content)
+		res.SetContentFormat(mediaType)
+		res.SetBody(bytes.NewReader(dataJson))
+		return res, nil
+	case message.AppCBOR, message.AppOcfCbor:
+		var v interface{}
+		err := json.Unmarshal(dataJson, &v)
```

Comment:
This seems a bit weird, first encode it to json on line 56, then decode it here and then encode it to cbor in `CreateResponseContent`. Is the encoding to json necessary when we have message.AppCBOR, message.AppOcfCbor?

Reply:
It seems that the cbor encoder fails on `thingDescription`, but succeeds on map decoded from the JSON, I'll add a note about it.

Reply:
thingDescription has implemented json marshaler/umarshaler interface (customize) which is not support by cbor.